src/Workspaces/Core/Portable/Formatting/Engine/AbstractFormatEngine.OperationApplier.cs

```diff
@@ -68,6 +68,11 @@ private bool ApplyDynamicSpacesOperation(AdjustSpacesOperation operation, int pa
                 // check whether current position is less than indentation
                 if (endColumnOfPreviousToken < indentation)
                 {
+                    if (_context.IsFormattingDisabled(pairIndex))
```

Comment:
things usually checked in one place
http://source.roslyn.io/#Microsoft.CodeAnalysis.CodeStyle/Formatting/Engine/AbstractFormatEngine.cs,429

is this didn't work out? if anchor and special (alignment) was an issue, we can add check there as well with new option in SuppressOption saying it is for pragma?